### ISSUE TITLE
Corrupted features in pretext generator

### ISSUE THREAD

**Issue by mortezamg63** [jsyoon0823/VIME#7]
Could you please let me know whether you use corruption (swap operation) before one-hot encoding or after one-hot encoding?

Currently, I encode categorical features to one-hot representation, then I feed the numerical features and one-hot encoded features to the neural network. So, the corruption is done on all numerical and one-hot encoded features. Should I use corruption before one-hot encoding?

**Comment by mortezamg63** [comment]
Also, it is mentioned in your paper that for categorical features, the cross-entropy loss is used. Do I need to use separate encoders for each categorical feature? or Do I need to use the output of one decoder  (reconstructed input) and split it to calculate loss (MSE for numerical features, cross-entropy for categorical features, and so on)?

**Comment by jsyoon0823** [comment]
Thank you for your interest in our paper and work.

1. For categorical features, we should swap first and then do one-hot encoding.
2. Yes. We need to use different loss functions for different variables (also different activation functions).

Thanks!

**Comment by mortezamg63** [comment]
Thanks for your response. Final questions. Do you set the mask for swapped categorical feature to one like the pic below?

![image](https://user-images.githubusercontent.com/15813546/174498729-9a28fb9d-1c5e-4e6a-afe1-69b59523f22b.png)

Also, do you use separate decoders for numeric and categorical features?  Does each categorical feature have a separate decoder? 
From MNIST dataset and shared code in this repo, it seems that all numerical features need one decoder. Right?
Could you let me know what activations do you use for different feature types?

**Comment by jsyoon0823** [comment]
1. For categorical features, I think either providing repeated masks (like the figure) or one mask per feature would similarly work. I use one mask per feature.
2. In MNIST, we do not need different decoders because all the features are continuous values. Therefore, we can use one decoder with same activation and loss function.
3. For categorical features (more than binary), we need separate decoder with softmax as the activation function and softmax cross entropy loss for that decoder.

**Comment by mortezamg63** [comment]
Thank you for your reply
